$NetBSD: patch-libgcc_config.host,v 1.3 2024/04/01 14:33:57 js Exp $

This is needed for arm64 to get Out-of-line LSE atomics linked into libgcc.
libsupc++ fails to link without this, and for arm to build.

Fix build on QNX.

--- libgcc/config.host.orig	2023-07-07 07:08:21.000000000 +0000
+++ libgcc/config.host
@@ -405,6 +405,7 @@ aarch64*-*-freebsd*)
 aarch64*-*-netbsd*)
 	extra_parts="$extra_parts crtfastmath.o"
 	tmake_file="${tmake_file} ${cpu_type}/t-aarch64"
+	tmake_file="${tmake_file} ${cpu_type}/t-lse"
 	tmake_file="${tmake_file} ${cpu_type}/t-softfp t-softfp t-crtfm"
 	md_unwind_header=aarch64/aarch64-unwind.h
 	;;
@@ -491,7 +492,8 @@ arm*-*-netbsdelf*)
 	case ${host} in
 	  arm*-*-netbsdelf-*eabi*)
 	    tmake_file="${tmake_file} arm/t-netbsd-eabi"
-	    unwind_header=config/arm/unwind-arm.h
+	    # GCC 7 vs NetBSD/eabi -> avoid arm unwinder
+	    #unwind_header=config/arm/unwind-arm.h
 	    ;;
 	  *)
 	    tmake_file="${tmake_file} arm/t-netbsd t-slibgcc-gld-nover"
@@ -797,7 +799,7 @@ i[34567]86-pc-msdosdjgpp*)
 	;;
 i[34567]86-*-lynxos*)
 	;;
-i[34567]86-*-nto-qnx*)
+i[34567]86-*-qnx*)
 	tmake_file="$tmake_file i386/t-nto t-libgcc-pic"
 	extra_parts=crtbegin.o
 	;;
